src/controls/Button.tsx

```diff
@@ -59,35 +55,35 @@ class ClickActionProperty extends ControlProperty<string> {
     return {
       elementToMount: fragment as any,
     };
-  }
-}
-
-export class Button extends Control {
-  private buttonElement: HTMLButtonElement;
-
-  @controlProperty(new TextAlignmentProperty((c: Button) => c.buttonElement))
-  public text: string;
-
-  @controlProperty(new TextFormattingProperty((c: Button) => c.buttonElement))
-  public textFormatting: Formatting;
+  },
+};
 
-  @controlProperty(new FontSizeProperty((c: Button) => c.buttonElement))
-  public fontSize: number;
-
-  @controlProperty(new TextContentProperty((c: Button) => c.buttonElement))
-  public textAlignment: string;
-
-  @controlProperty(new ClickActionProperty((c: Button) => c.buttonElement))
-  public clickScript: string;
-
-  protected initialize(): HTMLElement {
-    this.buttonElement = document.createElement('button');
-    return this.buttonElement;
-  }
+interface IButton extends IControlWithText {
+  clickScript: string;
+}
 
-  public get descriptor(): IControlDescriptor<Control> {
-    return buttonDescriptor;
-  }
+interface IButtonState {
+  root: HTMLElement;
 }
 
-export let buttonDescriptor = new ReflectionBasedDescriptor('button', Button);
+export const Button = createControlDefinition<IButton, IButtonState>({
+  id: 'button',
+  displayName: 'Button',
+  factory: function () {
```

Comment:
Ultimately this idea was founded on the idea that the "state" of an item was just a bag of elements, and the operations to be performed were defined via properties below.

Thus if you said:

```
let button = new Button()
```

the framework was creating new state for you, applying it to the object, and then the properties were defined to merely act on that state - again all by the framework.